dynamo-lib/src/cmr/dynamo/connection.clj

```diff
@@ -1,35 +1,89 @@
 (ns cmr.dynamo.connection
   "Contains functions for interacting with the DynamoDB storage instance"
-  (:require [cmr.common.config :as cfg :refer [defconfig]]
+  (:require [clojure.string :as str]
+            [cmr.common.config :as cfg :refer [defconfig]]
+            [cmr.common.services.errors :as errors]
+            [cmr.common.log :refer [debug error info trace warn]]
             [taoensso.faraday :as far]
             [cmr.dynamo.config :as dynamo-config]))
 
+(defn concept-revision-tuple->key
+  [tuple]
+  {:concept-id (first tuple) :revision-id (second tuple)})
+
+(defn provider-concept-revision-tuple->key
+  [tuple]
+  {:concept-id (nth tuple 1) :revision-id (nth tuple 2)})
+
+(defn find-params->dynamo-clause
+  ([params]
+   (find-params->dynamo-clause params false))
```

Comment:
as a suggestion, it has been said in other reviews that passing false in this way is slightly bad form instead of sending in something more descriptive like :use-or and then checking for that. Up to you, but false means so little.

Reply:
I actually pulled that code largely from elsewhere and then just adjusted the output for dynamo

Reply:
yea, it is a comment that came up before, I like the idea, it makes it easy to read, and something you can take advantage of since we are in clojure.